src/Sylius/Component/Core/Model/OrderItem.php

```diff
@@ -57,4 +57,45 @@ public function equals(BaseOrderItemInterface $item)
             && $item->getVariant() === $this->variant
         );
     }
+
+    /**
+     * Returns sum of neutral and non neutral tax adjustments on order item and total tax of units.
+     *
+     * {@inheritdoc}
+     */
+    public function getTaxTotal()
+    {
+        $taxTotal = 0;
+
+        foreach ($this->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT) as $taxAdjustment) {
+            $taxTotal += $taxAdjustment->getAmount();
+        }
+
+        foreach ($this->units as $unit) {
+            $taxTotal += $unit->getTaxTotal();
+        }
+
+        return $taxTotal;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDiscountedUnitPrice()
+    {
+        $orderItemDiscounts = 0;
+        foreach ($this->units as $unit) {
+            $orderItemDiscounts += $unit->getAdjustmentsTotal(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT);
```

Comment:
@peteward the discounts are applied at the `OrderItemUnit` level. This method should return discounted unit price e.g. `$1 discount on each T-Shirt`
* `getUnitPrice` = $10
* `getDiscountedUnitPrice` = $9

Proposed implementation:
```php
if ($this->units->isEmpty()) {
    return $this->unitPrice;
}

$unit = $this->units->first();
$unitDiscount = $unit->getNonNeutralAdjustmentsTotal(AdjustmentInterface::ORDER_ITEM_UNIT_PROMOTION_ADJUSTMENT);

return $this->unitPrice + $unitDiscount; // we add discount since it is negative
```

@Zales0123 we should introduce the `ORDER_ITEM_UNIT_PROMOTION` adjustment and use it instead of `ORDER_ITEM_PROMOTION`. Unit promotion adjustment should be equally distributed on all units, so there shouldn't be any unit with different discount. This type of adjustment should be always used for "per unit" discounts. But let's do this change in a separate PR.